common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -1094,6 +1094,48 @@ public static Geometry lineInterpolatePoint(Geometry geom, double fraction) {
     return geom.getFactory().createPoint(interPoint);
   }
 
+  public static double perimeter(Geometry geometry, boolean use_spheroid, boolean lenient) {
+    if (use_spheroid && geometry.getSRID() != 4326) {
+      if (lenient) {
+        return 0;
```

Comment:
I would prefer to assume SRID is 4326 if lenient is true. This is particularly useful because most people forget to set SRID for their 4326 geometries.